packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js

```diff
@@ -87,6 +94,20 @@ export default class QuanticGeneratedAnswer extends LightningElement {
     );
   };
 
+  buildHeadlessGeneratedAnswerController(engine) {
+    const storedGeneratedAnswerVisibility = sessionStorage?.getItem(
+      GENERATED_ANSWER_VISIBILITY_KEY
+    );
+    if (['true', 'false'].includes(storedGeneratedAnswerVisibility)) {
+      return this.headless.buildGeneratedAnswer(engine, {
+        initialState: {
+          isVisible: storedGeneratedAnswerVisibility === 'true' ? true : false,
+        },
+      });
+    }
+    return this.headless.buildGeneratedAnswer(engine);
```

Comment:
IMO storing a data object would made this code a little simpler.

Also, why not always pass the `initialState` when building the controller?

Reply:
I'm adding this logic to pass the `initiallState` only when valid values are being passed to the initial state to avoid having errors with the `setVisible` action in headless.

The scenario I'm trying to prevent with this is a scenario where the client is  manually updating the value stored in the session storage to an invalid value (any value other than `true` or `false`).
In this scenario instead of having a bunch of errors in the console coming from headless saying that it got an invalid value for the `setVisible` action, I'm, from the Quantic component, completely ignoring this invalid values, the Quantic component is taking the session storage in consideration only when it contains valid data. That's my logic here.

Reply:
Fair enough